### PR TITLE
feat: Scope.selectable_references()

### PR DESCRIPTION
**Pull request by barakalon**
references... selectables... referenced_selectables... selectable_references... 
The names might be getting confusing.

### CODE REVIEW

**Comment by barakalon** [comment]
Some refactoring:

- selectables -> sources
- referenced_selectables -> selected_sources
- columns, tables, derived_tables, subqueries, ctes -> *_nodes (plus this is searched for at access time... might be slightly less efficient, but it should be hard to screw up)
- references -> columns
- correlations -> columns_from_subqueries
- external_references -> columns_referencing_outer_sources
- selectable_references -> source_columns
-

**Comment by barakalon** [comment]
- columns_referencing_outer_sources -> external_columns
- Turns out I can just remove the column_nodes property - its only used by the columns property, so I just consolidated them.
- I removed all the _nodes suffixes.

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -637,7 +637,7 @@ def subquery(self, alias=None, copy=True):
         instance = _maybe_copy(self, copy)
         return Subquery(
             this=instance,
-            alias=to_identifier(alias),
+            alias=TableAlias(this=to_identifier(alias)),
```
Comment (tobymao) [inline sqlglot/expressions.py:640]: can you do alias_(alias, table=True)

**Review by tobymao — APPROVED** [review]

File: sqlglot/expressions.py (hunk 1 above)
Comment (barakalon) [reply sqlglot/expressions.py:640]: Mm I don't think so - we need to return a Subquery instance.

**Pull request merged**